### PR TITLE
RPP Test suite upgrade 3

### PR DESCRIPTION
**Pull request by r-abishek**
@kiritigowda @paveltc This PR is to be merged after #179, #180, #181, #213 - after merge conflict resolution from all of them.

This PR adds golden outputs for 11 tensor augmentations which account for majority of the files changed.

This PR introduces 3rd round of test suite changes:
- Makes necessary fixes to some functionalities to ensure QA test pass
- Removes testAllScripts.sh in both HOST and HIP backend - only one python script to run all variants
- Adds build flags in CMakeLists.txt to set AVX/SSE flags based on the system configuration
- Adds CPP calls in the runTests.py in both HOST and HIP backend
- Adds a new option ROI to the test suite to get the ROI coordinates as a list from the user

### CODE REVIEW

**Comment by kiritigowda** [comment]
@LakshmiKumar23 can you verify this PR?

**Comment by kiritigowda** [comment]
@r-abishek can you look at the file conflicts?

**Comment by r-abishek** [comment]
> @r-abishek this pr still has merge conflicts, can you try to resolve this?

#181 and #213 need to be merged into develop too, followed by a final test and merge conflict resolution for this PR.

**Review by rrawther — COMMENTED** [review]
why there are 101 file changes for this PR?

**Comment by LakshmiKumar23** [comment]
@kiritigowda Abishek mentioned to wait for PR181 and PR213 to merge before testing this

**Comment by kiritigowda** [comment]
@r-abishek - sync to fix merge conflicts and changelog required.

**Comment by kiritigowda** [comment]
@r-abishek sync to TOT

**Comment by r-abishek** [comment]
> @r-abishek sync to TOT

@kiritigowda Will initiate the TOT sync process on this PR once RICAP is merged into develop, else it may involve resolving more merge conflicts.

**Comment by kiritigowda** [comment]
> > @r-abishek sync to TOT
> 
> @kiritigowda Will initiate the TOT sync process on this PR once RICAP (#213 ) is merged into develop, else it may involve resolving more merge conflicts.

@r-abishek the RICAP PR is merged since the CI pass.

**Comment by kiritigowda** [comment]
@r-abishek is this ready to be merged?

**Comment by r-abishek** [comment]
> @r-abishek is this ready to be merged?

@kiritigowda Could you trigger the CI again? I want to ensure my fixes took effect

**Comment by kiritigowda** [comment]
@r-abishek - I can merge this once this passes - http://math-ci.amd.com/blue/organizations/jenkins/main%2FShort-GPUOpen%2Frpp/detail/PR-205/15/pipeline

**Comment by rrawther** [comment]
Hi @kiritigowda and @r-abishek: What is the status of this? Can we merge it

File: src/modules/cpu/kernel/resize.hpp
Comment (rrawther) [inline src/modules/cpu/kernel/resize.hpp:819]: why is this changed?

**Review by rrawther — APPROVED** [review]
@paveltc to test this PR
@kiritigowda Can you please check the test file changes with abishek. There are quite a few

File: src/modules/cpu/kernel/resize.hpp
Comment (r-abishek) [reply src/modules/cpu/kernel/resize.hpp:819]: Was a fix to match with QA test golden outputs - that are now in line with training convergence.

**Pull request merged**
